src/main/java/org/scijava/ui/behaviour/io/InputTriggerConfig.java

```diff
@@ -492,4 +492,41 @@ void addMap( final InputMap map, final String context )
 			}
 		}
 	}
+
+	/*
+	 * retrieving InputTriggerMaps from this InputTriggerConfig
+	 */
+
+	/**
+	 * Get a {@link InputTriggerMap} that represents the current setup
+	 * of the key-behaviour bindings relevant in the given context.
+	 *
+	 * @param context queried context
+	 * @return all bindings for the given context
+	 */
+	public InputTriggerMap createMapFor( final String context )
+	{
+		final InputTriggerMap map = new InputTriggerMap();
+		resetThisMapFor( map, context );
+		return map;
+	}
+
+	/**
+	 * Clear and set the {@link InputTriggerMap} such that it represents the
+	 * current setup of the key-behaviour bindings relevant in the given context.
+	 *
+	 * @param resetThisMap clear and set this map
+	 * @param context queried context
+	 */
+	public void resetThisMapFor(final InputTriggerMap resetThisMap, final String context )
+	{
+	    resetThisMap.clear();
+		for (String behaviourName : actionToInputsMap.keySet())
+		{
+			for (InputTrigger trigger : getInputs( behaviourName, context ))
+			{
+				resetThisMap.put( trigger, behaviourName );
+			}
+		}
+	}
```

Comment:
There is already functionality related to this.
First, this could be rewritten using an `InputTriggerAdder` which also takes care of corner cases (explicit `NOT_MAPPED` trigger, ...)
```java
	public void resetThisMapFor(final InputTriggerMap resetThisMap, final String context )
	{
		resetThisMap.clear();
		final InputTriggerAdder adder = inputTriggerAdder( resetThisMap, context );
		actionToInputsMap.keySet().forEach( adder::put );
	}
```

Second, there is `Behaviours.updateKeyConfig()` which provides very similar functionality.
https://github.com/scijava/ui-behaviour/blob/37d73656381bb698f455182b498b07fac5c4c459/src/main/java/org/scijava/ui/behaviour/util/Behaviours.java#L182-L223

You could *almost* rewrite `resetThisMapFor` as
```java
public void resetThisMapFor(final InputTriggerMap resetThisMap, final String context, final BehaviourMap behaviourMap )
{
	new Behaviours( resetThisMap, behaviourMap, null, context ).updateKeyConfig( this );
}
```
Note, the difference is the additional `behaviourMap` parameter. This will be used to decide which `behaviourName`s to retrieve from the `InputTriggerConfig`.

This is in contrast to your `resetThisMapFor`, which takes *all* `behaviourName`s that are defined in the `InputTriggerMap`. This will most likely work in the simple scenario where you just have a single `InputTriggerMap`/`BehaviourMap` pair, no chaining, blocking, etc.

You *could* replicate this as follows:
```java
public void resetThisMapFor(final InputTriggerMap resetThisMap, final String context)
{
	final BehaviourMap behaviourMap = new BehaviourMap();
	actionToInputsMap.forEach( ( action, inputs ) -> behaviourMap.put( action, new Behaviour() { /*dummy*/ } ) );
	new Behaviours( resetThisMap, behaviourMap, null, context ).updateKeyConfig( this );
}
```

However, I recommend that you use the actual `BehaviourMap` if you have it.

If you have a `Behaviours` instance which you use to add `Behaviour`s, then that manages an `InputTriggerMap`/`BehaviourMap` pair, already has a context and so on. In this case, I would try to keep this `Behaviours` instance around and simply do
```
behaviours.updateKeyConfig( theInputTriggerConfig );
```

In summary: Could you try to get the functionality you do via `createMapFor()` / `resetThisMapFor()` using the existing `updateKeyConfig` mechanism and report back? If that is feasible, I would not add `createMapFor()` and `resetThisMapFor()` to encourage `updateKeyConfig` as The Way&trade; to do updates.

Reply:
hi, thanks for the detailed explanation, I have again learnt something new and I think I understand this package a bit better now, thank you indeed for your patience -- you create always super generic SW, that is sometimes difficult for ordinary programmers when they come to it with zero knowledge of it, it took me a while to see it as a whole... the behaviour map, input trigger map and config, and the "binder" InputTriggerAdder... and obviously I only thought I'm seeing it all, for example, I missed totally the existence of Behaviours... and most of the utils...

also IDE got me here... i was not reading docs, should have, my bad, I know, I relied instead on method names only, these can be sometimes tricky, need their context... here, it did not occur to me that `Behaviours.updateKeyConfig` actually updates Map and leaves the Config (its parameter) untouched

anyway,  please drop the commit of mine from this PR
I confirm, it was easy to use the updateKeyConfig mechanism: https://github.com/scenerygraphics/scenery/pull/324/commits/ed025a74a8189b1279c2646cc17592caa6de00d6
(except that it always creates a new object... that's failure on the scenery side, it could have been caching the aux objects next to the three maps)

Reply:
> I missed totally the existence of Behaviours... and most of the utils...

Yeah, this is not very discoverable. Basically the "preferred way of wiring up everything" has evolved over time (and is still evolving). With Mastodon it has crystallized that a good pattern is to maintain the `Behaviours` (and sibling `Actions`) as the "handles" for adding/updating input/action maps. (Older stuff , e.g., BDV, doesn't match that 100% yet but should be updated eventually.)

> also IDE got me here... i was not reading docs, should have, my bad, I know, I relied instead on method names only, these can be sometimes tricky, need their context... here, it did not occur to me that `Behaviours.updateKeyConfig` actually updates Map and leaves the Config (its parameter) untouched

Hmm, yes, that is very unfortunately named... :-( `Behaviours.updateWithKeyConfig(...)`, or maybe just `updateWith(...)` would be better. Do you have any other ideas?

Reply:
aaaa, there's even `Actions`... I should really be paying more attention to "satellite" stuff around me, good know this. THANK YOU.

regarding the method name, what about `Behaviours.updateOwnMapFromThisConfig(...)` or perhaps without the pronoun "Own", or with? "updateIts" , "updateInternal", but the word Map probably would be worthwhile to have unless the method changes the whole object (like also the behaviours... )